sherpa/plot/tests/test_plot_notebook.py

```diff
@@ -34,16 +34,25 @@
 from sherpa.models.basic import Const1D, Gauss2D, Polynom1D
 from sherpa.stats import Chi2
 
-
-def plot_backend_is(name):
-    """Check we have the specified plot backend"""
-    return plot.backend.name == name
+# This not the most elegant solution, but it makes sense to
+# have an independent check here and not rely on what is
+# done in the sherpa.plot.__init__ module, so that the tests
+# stay independent of that particular implementation.
+try:
+    from sherpa.plot.pylab_backend import PylabBackend
+    HAS_PYLAB = True
+except ModuleNotFoundError:
+    HAS_PYLAB = False
 
 
 def check_empty(r, summary, nsummary=0):
     """Is this an 'empty' response?"""
 
-    if plot_backend_is("pylab"):
+    # If not HAS_PYLAB, then the symbol PylabBackend is not defined
+    # and test collection will fail. So, we short-circuit that
+    # such that PylabBackend will only  be needed it
+    # HAS_PYLAB is True.
```

Comment:
It's not strictly necessary. I just thought it would help me next time I look at that code to explain why I do all the HAS_PYLAB stuff and not just that simple `isinstance` that makes up the second part of the test. I'll reword.